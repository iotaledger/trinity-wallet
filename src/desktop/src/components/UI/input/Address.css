.address-input {
    display: flex;
    justify-content: space-between;
    width: 100%;
    max-width: 512px;
    margin: 0 auto 30px;
    > div {
        display: block;
        position: relative;
        width: calc(100% - 70px);
    }
    input[type='text'] {
        font-family: 'Lato';
        font-weight: 200;
        font-size: var(--font-size-default);
        line-height: 26px;
        width: 100%;
        border: none;
        background: none;
        color: #fff;
        padding: 4px 0;
        font-size: 18px;
        resize: none;
        &:focus {
            outline: none;
        }
    }
    button {
        padding: 0 8px;
        height: 32px;
        min-width: 0px;
        font-size: 11px;
        margin: 1px 0 0;
        line-height: 18px;
        img {
            width: 14px;
            vertical-align: middle;
            margin: -2px 4px 0 0;
        }
    }
    small {
        position: absolute;
        bottom: 1px;
        left: 0px;
        display: block;
        width: 100%;
        text-align: left;
        &:before,
        &:after {
            position: absolute;
            content: '';
            width: 100%;
            height: 1px;
            background: white;
            top: 0px;
            left: 0px;
        }
        &:after {
            max-width: 0px;
            background: var(--color-yellow);
            transition: all 0.2s ease-out;
        }
    }
    input[type='text']:focus + small:after {
        max-width: 100%;
    }
}

.qrScanner {
    width: 640px;
    max-width: 100%;
    margin: auto;
    text-align: center;
    video {
        transform: rotateY(180deg);
        max-width: 100%;
        margin-bottom: 50px;
        border: 2px solid #fff;
    }
}
