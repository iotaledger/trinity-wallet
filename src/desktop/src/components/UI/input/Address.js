import React from 'react';
import QrReader from 'react-qr-reader';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { ADDRESS_LENGTH } from 'libs/util';
import css from './Input.css';
import Modal from 'components/UI/Modal';

import Camera from 'images/camera.png';

export default class AddressInput extends React.PureComponent {
    static propTypes = {
        address: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        closeLabel: PropTypes.string.isRequired,
    };

    state = {
        showScanner: false,
    };

    openScanner = e => {
        e.preventDefault();
        this.setState(() => ({
            showScanner: true,
        }));
    };

    closeScanner = e => {
        e.preventDefault();
        this.setState(() => ({
            showScanner: false,
        }));
    };

    onScanEvent = address => {
        if (address !== null) {
            this.setState(() => ({
                showScanner: false,
            }));
            this.props.onChange(address);
        }
    };

    onScanError = err => {
        console.log(err);
    };

    render() {
        const { address, label, closeLabel } = this.props;
        const { showScanner } = this.state;

        return (
            <div className={css.input}>
                <fieldset>
                    <a onClick={this.openScanner}>
                        <img src={Camera} alt="" />
                    </a>
                    <input
                        type="text"
                        value={address}
                        onChange={e => this.props.onChange(e.target.value)}
                        maxLength={ADDRESS_LENGTH}
                    />
                    <small>{label}</small>
                </fieldset>
                {showScanner && (
                    <Modal isOpen={true} onStateChange={showScanner => this.setState({ showScanner })} hideCloseButton>
                        <div className={css.qrScanner}>
                            <QrReader delay={350} onError={this.onScanError} onScan={this.onScanEvent} />
                            <Button type="button" onClick={this.closeScanner} variant="cta">
                                {closeLabel}
                            </Button>
                        </div>
                    </Modal>
                )}
            </div>
        );
    }
}
