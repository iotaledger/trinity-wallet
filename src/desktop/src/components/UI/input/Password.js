import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import css from './Input.css';

import IconEye from 'images/eye.png';

export default class PasswordInput extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string,
        value: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
    };

    state = {
        type: 'password',
    };

    toggleVisibility = e => {
        e.preventDefault();
        this.setState(state => ({
            type: state.type === 'password' ? 'text' : 'password',
        }));
    };

    render() {
        const { className, label, value } = this.props;
        const { type } = this.state;
        return (
            <div className={css.input}>
                <fieldset>
                    <a className={type === 'text' ? css.strike : null} onClick={this.toggleVisibility}>
                        <img src={IconEye} alt="" />
                    </a>
                    <input type={type} value={value} onChange={e => this.props.onChange(e.target.value)} />
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
