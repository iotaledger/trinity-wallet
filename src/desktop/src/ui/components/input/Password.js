import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'ui/components/Icon';
import css from './input.css';

/**
 * Password input component
 */
export default class PasswordInput extends React.PureComponent {
    static propTypes = {
        /** Current password value */
        value: PropTypes.string.isRequired,
        /** Password input label */
        label: PropTypes.string.isRequired,
        /** Password change event function
         * @param {string} value - Current password value
         */
        onChange: PropTypes.func.isRequired,
    };

    state = {
        hidden: true,
    };

    setVisibility = () => {
        this.setState({
            hidden: !this.state.hidden,
        });
    };

    render() {
        const { label, value, onChange } = this.props;
        const { hidden } = this.state;
        return (
            <div className={css.input}>
                <fieldset>
                    <a className={hidden ? css.strike : null} onClick={this.setVisibility}>
                        <Icon icon="eye" size={16} />
                    </a>
                    <input
                        type={hidden ? 'password' : 'text'}
                        value={value}
                        onChange={(e) => onChange(e.target.value)}
                    />
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
