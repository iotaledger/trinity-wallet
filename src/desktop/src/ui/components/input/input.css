.input {
    display: block;
    width: 100%;
    padding: 25px 0 30px;
    text-align: right;
    fieldset {
        padding: 0px;
        margin: 0px;
        border: none;
        position: relative;
        text-align: left;
        input,
        textarea,
        select {
            font-family: 'Lato';
            font-weight: 200;
            font-size: var(--font-size-default);
            line-height: 26px;
            width: 100%;
            border: 1px solid var(--input-border);
            background: var(--input-bg);
            color: var(--input);
            padding: 6px 12px;
            resize: none;
            -webkit-appearance: none;
            border-radius: 4px;
            &:focus {
                outline: none;
            }
            &:disabled {
                pointer-events: none;
                opacity: 0.5;
            }
        }
        select {
            position: relative;
            &:after {
                content: '';
                display: block;
                width: 0;
                height: 0;
                border-style: solid;
                border-width: 6px 4px 0 4px;
                border-color: var(--body) transparent transparent transparent;
                position: absolute;
                right: 12px;
                top: calc(50% - 2px);
                opacity: 0.6;
                pointer-event: none;
            }
        }
        small {
            position: absolute;
            white-space: nowrap;
            left: 0px;
            top: -20px;
            font-size: var(--font-size-xs);
            color: var(--label);
            text-transform: uppercase;
            transition: color 0.2s ease-out;
        }
        input:focus + small,
        textarea:focus + small,
        select:focus + small {
            color: var(--label-hover);
        }
        a {
            display: flex;
            position: absolute;
            align-items: center;
            justify-content: center;
            position: absolute;
            height: 100%;
            min-width: 35px;
            top: 0px;
            right: 0px;
            cursor: pointer;
            user-select: none;
            color: var(--input-alt);
            &:before {
                content: '';
                position: absolute;
                width: 2px;
                height: 22px;
                top: 50%;
                left: -2px;
                border-radius: 4px;
                background: var(--input-alt);
                transform: translate(0, -50%);
            }
            > strong {
                font-size: 11px;
                font-weight: 200;
                > span {
                    margin-left: 4px;
                }
            }
            &:hover {
                color: var(--input);
            }
        }
        a + input {
            padding-right: 40px;
        }
    }
}

.info {
    display: block;
    float: right;
    font-size: 11px;
    color: var(--input);
    line-height: 14px;
    padding: 4px 8px 2px;
    border-radius: 0 0 4px 4px;
    position: relative;
    background: var(--input-bg);
    margin: -1px 38px 0 0;
    border: 1px solid var(--input-border);
    border-top: none;
}

.checkbox {
    display: block;
    float: right;
    margin: 12px 3px 0 auto;
    font-size: 10px;
    line-height: 14px;
    padding: 0 38px 2px 0;
    position: relative;
    text-transform: uppercase;
    cursor: pointer;
    &:after {
        content: '';
        width: 28px;
        height: 15px;
        position: absolute;
        top: 0px;
        right: 1px;
        border-radius: 15px;
        border: 1px solid var(--body);
    }
    &:before {
        content: '';
        width: 9px;
        height: 9px;
        position: absolute;
        top: 3px;
        right: 16px;
        border-radius: 9px;
        border: 1px solid var(--body);
        transition: all 0.2s cubic-bezier(0, 0, 0.2, 1);
    }
    &.on {
        color: var(--primary);
        &:before {
            border-color: var(--primary);
            background: var(--primary);
            right: 4px;
        }
    }
}

.strike {
    &:after {
        content: '';
        position: absolute;
        width: 1px;
        height: 16px;
        border-radius: 2px;
        background: var(--input-alt);
        transform: rotate(45deg);
    }
    &:hover:after {
        background: var(--input);
    }
}

.down {
    position: relative;
    opacity: 1;
    z-index: 2;
    pointer-events: none;
    &:before {
        display: none;
    }
    &:after {
        content: '';
        display: block;
        width: 0;
        height: 0;
        border-style: solid;
        border-width: 6px 4px 0 4px;
        border-color: var(--input) transparent transparent transparent;
        position: absolute;
        right: 12px;
        top: calc(50% - 2px);
    }
}

.qrScanner {
    display: block;
    margin: 0 auto;
    width: 320px;
    text-align: center;
    video {
        transform: rotateY(180deg);
        max-width: 100%;
        margin-bottom: 50px;
        border: 2px solid var(--input);
    }
    > section {
        margin-bottom: 40px;
    }
}
