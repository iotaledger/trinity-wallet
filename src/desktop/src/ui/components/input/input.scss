.input {
    display: block;
    width: 100%;
    max-width: 520px;
    margin: 0 auto;
    padding: 25px 0 30px;
    transition: opacity 0.4s cubic-bezier(0.165, 0.84, 0.44, 1);
    position: relative;

    fieldset {
        border: none;
        position: relative;
        input,
        textarea,
        select,
        .editable {
            font-family: 'SourceSansPro';
            font-weight: 300;
            font-size: var(--font-size-default);
            line-height: 46px;
            width: 100%;
            border: 1px solid var(--input-border);
            background: var(--input-bg);
            color: var(--input);
            padding: 0 20px 0 14px;
            resize: none;
            -webkit-appearance: none;
            border-radius: 4px;
            transition: border 0.4s cubic-bezier(0.165, 0.84, 0.44, 1);
            &:focus {
                outline: none;
                border-color: var(--input-focus);
            }
            &:disabled {
                pointer-events: none;
                opacity: 0.5;
            }
        }
        .editable {
            display: block;
            font-family: 'SourceCodePro';
            white-space: nowrap;
            overflow: hidden;
            height: 46px;
            text-align: left;
            max-width: 520px;
            padding: 0 40px 0 0;
            > div {
                padding: 0 20px;
                overflow: hidden;

                span {
                    display: inline-block;
                    position: relative;
                    height: 100%;
                    &:before {
                        display: inline-block;
                        position: absolute;
                        left: 0px;
                        content: attr(data-letter);
                        pointer-events: none;
                        line-height: 44px;
                    }
                }
            }
        }
        select {
            position: relative;
            &:after {
                content: '';
                display: block;
                width: 0;
                height: 0;
                border-style: solid;
                border-width: 6px 4px 0 4px;
                border-color: var(--input-alt) transparent transparent transparent;
                position: absolute;
                right: 12px;
                top: calc(50% - 2px);
                pointer-event: none;
            }
        }
        small {
            position: absolute;
            white-space: nowrap;
            left: 0px;
            top: -24px;
            color: var(--label);
            text-transform: uppercase;
            transition: color 0.2s ease-out;
            font-size: 13px;
            letter-spacing: 0.2px;
        }
        input:focus + small,
        textarea:focus + small,
        select:focus + small {
            color: var(--label-hover);
        }
        a {
            display: flex;
            position: absolute;
            align-items: center;
            justify-content: center;
            position: absolute;
            height: 100%;
            min-width: 35px;
            top: 0px;
            right: 0px;
            cursor: pointer;
            user-select: none;
            color: var(--input-alt);
            &:before {
                content: '';
                position: absolute;
                width: 2px;
                height: 22px;
                top: 50%;
                left: -2px;
                border-radius: 4px;
                background: var(--input-alt);
                transform: translate(0, -50%);
            }
            > strong {
                font-size: 11px;
                > span {
                    margin-left: 4px;
                }
            }
            &:hover {
                color: var(--input);
            }
            &.right {
                left: 0px;
                right: auto;
                background: red;
                z-index: 2;
                width: 42px;
                background: var(--input-bg);
                border-radius: 4px 0 0 4px;
                border: 1px solid var(--input-border);
                border-right: none;
                &:before {
                    display: none;
                }
            }
        }
        p {
            display: block;
            position: absolute;
            top: 0px;
            right: 0px;
            right: 50px;
            line-height: 38px;
            white-space: nowrap;
            color: var(--input-alt);
            pointer-events: none;
        }
        a + input {
            padding-right: 40px;
        }
    }

    h5 {
        display: block;
        width: 100%;
        border: 1px dashed var(--body-alt);
        font-size: 12px;
        line-height: 12px;
        margin: 32px 0 0 0;
        padding: 24px 0;
        font-size: 14px;
        text-align: center;
        cursor: pointer;

        svg {
            display: block;
            margin: 0 auto 20px;
            pointer-events: none;
        }
        span {
            display: block;
            pointer-events: none;
            line-height: 18px;
        }
    }

    > fieldset > strong {
        display: none;
        position: absolute;
        top: 58px;
        left: 8px;
        text-align: left;
        pointer-events: none;
        font-weight: 300;
        font-size: 12px;
        line-height: 12px;
        color: var(--body);
        span {
            float: left;
            margin: -1px 4px 0 0;
        }
    }

    &.disabled {
        opacity: 0.25;
        pointer-events: none;
    }

    &.padded {
        margin-bottom: 18px;
    }

    &.seed {
        fieldset .editable {
            padding: 0 32px 0 21px;
        }
    }

    &.capsLock {
        input:focus + small + strong {
            display: block;
        }
        > :before {
            content: '';
            display: block;
            position: absolute;
            top: 1px;
            right: 40px;
            width: 46px;
            height: 46px;
            background: var(--input-bg);
        }
    }
}

.info {
    display: block;
    position: absolute;
    top: 71px;
    right: -5px;
    font-size: 11px;
    color: var(--input);
    line-height: 14px;
    padding: 4px 8px 2px;
    border-radius: 0 0 4px 4px;
    background: var(--input-bg);
    margin: -1px 38px 0 0;
    border: 1px solid var(--input-border);
    border-top: none;
}

.checkbox {
    display: block;
    position: absolute;
    bottom: -10px;
    right: 0px;
    float: right;
    font-size: 10px;
    line-height: 14px;
    padding: 0 38px 2px 0;
    text-transform: uppercase;
    cursor: pointer;
    &:after {
        content: '';
        width: 28px;
        height: 15px;
        position: absolute;
        top: 0px;
        right: 1px;
        border-radius: 15px;
        border: 1px solid var(--body);
    }
    &:before {
        content: '';
        width: 9px;
        height: 9px;
        position: absolute;
        top: 3px;
        right: 16px;
        border-radius: 9px;
        border: 1px solid var(--body);
        transition: all 0.2s cubic-bezier(0, 0, 0.2, 1);
    }
    &.on {
        color: var(--primary);
        &:before {
            border-color: var(--primary);
            background: var(--primary);
            right: 4px;
        }
    }
}

.dropdown {
    position: absolute;
    top: 0;
    left: 0px;
    width: 100%;
    padding: 38px 0 0 0;
    z-index: 2;
    max-height: 40px;
    overflow: hidden;
    li {
        display: block;
        list-style: none;
        padding-left: 8px;
        width: 100%;
        text-align: left;
        line-height: 28px;
        background: var(--input-bg);
        color: var(--input-alt);
        &.selected {
            color: var(--input);
        }
        &:last-of-type {
            padding-bottom: 8px;
            border-radius: 0 0 4px 4px;
        }
        &:hover {
            color: var(--input);
        }
    }
    &:hover {
        max-height: none;
    }
}

.strike {
    &:after {
        content: '';
        position: absolute;
        width: 1px;
        height: 16px;
        border-radius: 2px;
        background: var(--input-alt);
        transform: rotate(45deg);
    }
    &:hover:after {
        background: var(--input);
    }
}

.down {
    position: relative;
    z-index: 2;
    pointer-events: none;
    &:before {
        display: none;
    }
    &:after {
        content: '';
        display: block;
        width: 0;
        height: 0;
        border-style: solid;
        border-width: 6px 4px 0 4px;
        border-color: var(--input) transparent transparent transparent;
        position: absolute;
        right: 12px;
        top: calc(50% - 2px);
    }
}

.qrScanner {
    display: block;
    margin: 0 auto;
    width: 320px;
    text-align: center;
    video {
        transform: rotateY(180deg);
        max-width: 100%;
        margin-bottom: 50px;
        border: 2px solid var(--input);
    }
    > section {
        margin-bottom: 40px;
    }
}

.score {
    display: flex;
    position: absolute;
    top: -17px;
    right: 0px;
    span {
        display: block;
        height: 4px;
        width: 50px;
        background: var(--body-alt);
        margin-left: 2px;
    }
    &[data-strength='1'] {
        span:nth-child(1) {
            background: var(--negative);
        }
    }
    &[data-strength='3'],
    &[data-strength='2'],
    &[data-strength='1'] {
        span:nth-child(1),
        span:nth-child(2) {
            background: var(--negative);
        }
    }
    &[data-strength='4'] {
        span {
            background: var(--positive);
        }
    }
}

.hint {
    position: absolute;
    bottom: -28px;
    right: 0;
    font-size: 14px;
}

.valid {
    position: absolute;
    top: 10px;
    right: -40px;
    opacity: 0.25;
    &.isValid {
        opacity: 1;
    }
}
