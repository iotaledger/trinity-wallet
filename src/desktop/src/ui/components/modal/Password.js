import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { connect } from 'react-redux';

import { generateAlert } from 'actions/alerts';

import Password from 'ui/components/input/Password';
import Button from 'ui/components/Button';
import Modal from 'ui/components/modal/Modal';

import { getVault } from 'libs/crypto';

/**
 * Password confirmation dialog component
 */

class ModalPassword extends PureComponent {
    static propTypes = {
        /** Password window content */
        content: PropTypes.object.isRequired,
        /** Password window type */
        category: PropTypes.oneOf(['primary', 'secondary', 'positive', 'negative', 'highlight', 'extra']),
        /** Dialog visibility state */
        isOpen: PropTypes.bool,
        /** Should the dialog be without a cancel option */
        isForced: PropTypes.bool,
        /** Modal inline style state */
        inline: PropTypes.bool,
        /** On dialog close event */
        onClose: PropTypes.func,
        /** On correct password entered event
         * @param {String} Password - Entered password plain text
         * @param {Object} Vault - Vault content
         */
        onSuccess: PropTypes.func.isRequired,
        /** Create a notification message
         * @param {String} type - notification type - success, error
         * @param {String} title - notification title
         * @param {String} text - notification explanation
         * @ignore
         */
        generateAlert: PropTypes.func.isRequired,
        /** Translation helper
         * @param {String} translationString - Locale string identifier to be translated
         * @ignore
         */
        t: PropTypes.func.isRequired,
    };

    state = {
        password: '',
    };

    componentWillReceiveProps(nextProps) {
        if (this.props.isOpen !== nextProps.isOpen) {
            this.setState({
                password: '',
            });
        }
    }

    onSubmit(e) {
        const { password } = this.state;
        const { onSuccess, generateAlert, t } = this.props;

        e.preventDefault();

        let vault = null;

        try {
            vault = getVault(password);
        } catch (err) {
            generateAlert(
                'error',
                t('changePassword:incorrectPassword'),
                t('changePassword:incorrectPasswordExplanation'),
            );
            return;
        }

        if (vault) {
            onSuccess(password, vault);
        }
    }

    render() {
        const { content, category, isOpen, isForced, inline, onClose, t } = this.props;
        const { password } = this.state;

        return (
            <Modal variant="confirm" inline={inline} isOpen={isOpen} isForced={isForced} onClose={() => onClose()}>
                {content.title ? <h1 className={category ? category : null}>{content.title}</h1> : null}
                {content.message ? <p>{content.message}</p> : null}
                <form onSubmit={(e) => this.onSubmit(e)}>
                    <Password
                        value={password}
                        label={t('password')}
                        onChange={(value) => this.setState({ password: value })}
                    />
                    <fieldset>
                        {!isForced ? (
                            <Button onClick={() => onClose()} variant="secondary">
                                {t('cancel')}
                            </Button>
                        ) : null}
                        <Button type="submit" variant={category ? category : 'positive'}>
                            {content.confirm ? content.confirm : t('ok')}
                        </Button>
                    </fieldset>
                </form>
            </Modal>
        );
    }
}

const mapDispatchToProps = {
    generateAlert,
};

export default connect(null, mapDispatchToProps)(translate()(ModalPassword));
