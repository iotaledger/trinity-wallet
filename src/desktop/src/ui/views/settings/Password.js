import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { connect } from 'react-redux';

import { showError, showNotification } from 'actions/notifications';

import { isValidPassword } from 'libs/util';
import { getSecurelyPersistedSeeds, securelyPersistSeeds } from 'libs/storage';

import Password from 'components/UI/input/Password';
import Button from 'ui/components/Button';

/**
 * User account password change component
 */
class SetPassword extends PureComponent {
    static propTypes = {
        /* Error helper function
         * @param {Object} error notification content
         * @ignore
         */
        showError: PropTypes.func.isRequired,
        /* Notification helper function
         * @param {Object} success notification content
         * @ignore
         */
        showNotification: PropTypes.func.isRequired,
        /* Translation helper
         * @param {String} locale identifier to be translated
         * @ignore
         */
        t: PropTypes.func.isRequired,
    };

    state = {
        passwordCurrent: '',
        passwordNew: '',
        passwordConfirm: '',
    };

    changePassword = () => {
        const { passwordCurrent, passwordNew, passwordConfirm } = this.state;
        const { showError, showNotification, t } = this.props;

        if (passwordNew !== passwordConfirm) {
            showError({
                title: t('changePassword:passwordsDoNotMatch'),
                text: t('changePassword:passwordsDoNotMatchExplanation'),
            });
            return;
        }

        if (!isValidPassword(passwordNew)) {
            showError({
                title: t('changePassword:passwordTooShort'),
                text: t('changePassword:passwordTooShortExplanation'),
            });
            return;
        }

        try {
            const seeds = getSecurelyPersistedSeeds(passwordCurrent);

            securelyPersistSeeds(passwordNew, seeds);

            this.setState({
                passwordCurrent: '',
                passwordNew: '',
                passwordConfirm: '',
            });

            showNotification({
                title: t('changePassword:passwordUpdated'),
                text: t('changePassword:passwordUpdatedExplanation'),
            });
        } catch (err) {
            showError({
                title: t('changePassword:incorrectPassword'),
                text: t('changePassword:incorrectPasswordExplanation'),
            });
            return;
        }
    };

    render() {
        const { t } = this.props;
        const { passwordCurrent, passwordNew, passwordConfirm } = this.state;

        return (
            <div>
                <Password
                    value={passwordCurrent}
                    label={t('changePassword:currentPassword')}
                    onChange={(value) => this.setState({ passwordCurrent: value })}
                />
                <Password
                    value={passwordNew}
                    label={t('changePassword:newPassword')}
                    onChange={(value) => this.setState({ passwordNew: value })}
                />
                <Password
                    value={passwordConfirm}
                    label={t('changePassword:confirmPassword')}
                    onChange={(value) => this.setState({ passwordConfirm: value })}
                />

                <Button onClick={() => this.changePassword()}>{t('settings:changePassword')}</Button>
            </div>
        );
    }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = {
    showError,
    showNotification,
};

export default translate()(connect(mapStateToProps, mapDispatchToProps)(SetPassword));
