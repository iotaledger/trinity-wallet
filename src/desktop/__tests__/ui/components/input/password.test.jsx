import React from 'react';
import { shallow } from 'enzyme';

import { PasswordComponent as Password } from 'ui/components/input/Password';

const props = {
    value: 'LoremIpsumDolorSitAmet',
    label: 'Bar',
    showScore: true,
    onChange: jest.fn(),
    t: (str) => str,
};

describe('Password component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Password {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Input value', () => {
        const wrapper = shallow(<Password {...props} />);

        expect(wrapper.find('input').props().value).toEqual('LoremIpsumDolorSitAmet');
    });

    test('Input change callback', () => {
        const wrapper = shallow(<Password {...props} />);

        wrapper.find('input').simulate('change', { target: { value: 'FooBar' } });
        expect(props.onChange).toHaveBeenLastCalledWith('FooBar');
    });

    test('Password score display', () => {
        const testValues = ['Foo', 'FooBar', 'FooBarBaz', 'FooBarBazF', 'FooBarBazFizz'];

        testValues.forEach((testValue, index) => {
            const wrapper = shallow(<Password {...props} value={testValue} />);
            expect(wrapper.find('div.score').props()['data-strength']).toEqual(index);
        });
    });

    test('Password valid input display', () => {
        const validProps = [
            {
                value: 'FooBarBazFizz',
            },
            {
                value: 'FooBarBazFizz',
                match: 'FooBarBazFizz',
            },
        ];

        const invalidProps = [
            {
                value: 'FooBar',
            },
            {
                value: 'FooBar',
                match: 'FooBar',
            },
            {
                value: 'FooBarBazFizz',
                match: 'FooBarBazFuzz',
            },
        ];

        validProps.forEach((testProps) => {
            const wrapper = shallow(<Password {...props} {...testProps} showValid />);
            expect(wrapper.find('.isValid')).toHaveLength(1);
        });

        invalidProps.forEach((testProps) => {
            const wrapper = shallow(<Password {...props} {...testProps} showValid />);
            expect(wrapper.find('.isValid')).toHaveLength(0);
        });
    });

    test('Input label', () => {
        const wrapper = shallow(<Password {...props} />);

        expect(wrapper.find('small').text()).toEqual('Bar');
    });

    test('Input disabled state', () => {
        const mockProps = Object.assign({}, props, { disabled: true });
        const wrapper = shallow(<Password {...mockProps} />);

        expect(wrapper.hasClass('disabled')).toBeTruthy();
    });
});
