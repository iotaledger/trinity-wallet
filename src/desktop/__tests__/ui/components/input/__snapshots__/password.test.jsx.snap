// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password component Render the component 1`] = `
<div
  className="input padded"
>
  <fieldset>
    <a
      className="strike"
      onClick={[Function]}
    >
      <Icon
        icon="eye"
        size={16}
      />
    </a>
    <input
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="LoremIpsumDolorSitAmet"
    />
    <small>
      Bar
    </small>
    <strong>
      <Icon
        icon="attention"
        size={14}
      />
      capsLockIsOn
    </strong>
    <div
      className="score"
      data-strength={4}
    >
      <span />
      <span />
      <span />
    </div>
  </fieldset>
</div>
`;
